server/events/vcs/github_client.go

```diff
@@ -244,6 +244,12 @@ func (g *GithubClient) HidePrevCommandComments(repo models.Repo, pullNum int, co
 		if !strings.Contains(firstLine, strings.ToLower(command)) {
 			continue
 		}
+
+		// If dir was specified, skip processing comments that don't contain the dir in the first line
+		if dir != "" && !strings.Contains(firstLine, strings.ToLower(dir)) {
```

Comment:
My one concern is that this is a bit broad might lead to false positives (for example, if users have a directory called `workspace`, that will match `Ran Plan for dir: 'stack1' workspace: 'default'"`). One option might just be to look for `"'" + strings.ToLower(dir) + "'"`? Still would have some false positives, but would reduce them.

Reply:
I did consider extending the string to match, but the rest of the string, including the backticks is defined in the markdown templates, which can be user customised,  ref: https://www.runatlantis.io/docs/server-configuration.html#markdown-template-overrides-dir, so is not guaranteed to be in the same format.

Example: https://github.com/runatlantis/atlantis/blob/6123e2bb97e287b010767070753ee87804aaca7a/server/events/templates/single_project_plan_success.tmpl#L3

Reply:
Yeah that makes sense. We probably want to err on the side of *not* hiding comments anyway so I think this is fine.